olp-cpp-sdk-core/include/olp/core/client/OlpClient.h

```diff
@@ -127,6 +182,13 @@ class CORE_API OlpClient {
       const std::shared_ptr<std::vector<unsigned char>>& post_body,
       const std::string& content_type) const;
 
+  std::shared_ptr<http::NetworkRequest> CreateRequest(
+      const std::string& path, olp::http::NetworkRequest::HttpVerb method,
+      const std::multimap<std::string, std::string>& query_params,
+      const std::multimap<std::string, std::string>& header_params,
+      const std::shared_ptr<std::vector<unsigned char>>& post_body,
```

Comment:
Same here, pass by copy and use `RequestBodyType` if possible.

Reply:
I'll make it in a follow-up commit